pkg/ext/stores/tokens/tokens.go

```diff
@@ -273,6 +273,64 @@ func (t *Store) Create(
 	return t.create(ctx, objToken, options)
 }
 
+// DeleteCollection implements [rest.CollectionDeleter]
+func (t *Store) DeleteCollection(
+	ctx context.Context,
+	deleteValidation rest.ValidateObjectFunc,
+	options *metav1.DeleteOptions,
+	listOptions *metainternalversion.ListOptions,
+) (runtime.Object, error) {
+	userInfo, fullAccess, _, err := t.auth.UserName(ctx, &t.SystemStore, "delete")
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("error getting user info: %w", err))
+	}
+
+	lOptions, err := extcore.ConvertListOptions(listOptions)
+	if err != nil {
+		return nil, apierrors.NewInternalError(err)
+	}
+	// Non-system requests always filter the tokens down to those of the current user.
+	// Merge our own selection request (user match!) into the caller's demands
+	localOptions, err := ListOptionMerge(fullAccess, userInfo.GetName(), lOptions)
+	if err != nil {
+		return nil, apierrors.NewInternalError(fmt.Errorf("failed to process list options: %w", err))
+	}
+
+	secrets, err := t.secretClient.List(TokenNamespace, localOptions)
+	if err != nil {
+		if apierrors.IsResourceExpired(err) || apierrors.IsGone(err) { // Continue token expired.
+			return nil, apierrors.NewResourceExpired(err.Error())
+		}
+		return nil, apierrors.NewInternalError(fmt.Errorf("failed to list tokens: %w", err))
+	}
+
+	list := &ext.TokenList{
+		ListMeta: metav1.ListMeta{
+			Continue:           secrets.Continue,
+			ResourceVersion:    secrets.ResourceVersion,
+			RemainingItemCount: secrets.RemainingItemCount,
+		},
+		Items: make([]ext.Token, 0, len(secrets.Items)),
+	}
+
+	for _, secret := range secrets.Items {
+		obj, _, err := t.deleteToken(ctx, &secret, deleteValidation, options)
+		if err != nil {
+			return nil, apierrors.NewInternalError(fmt.Errorf("error deleting token %s: %w",
+				secret.Name, err))
+		}
+
+		token, ok := obj.(*ext.Token)
+		if !ok { // Sanity check.
+			return nil, apierrors.NewInternalError(fmt.Errorf("expected token object, got %T", obj))
+		}
```

Comment:
This goes with changing the return type of `deleteToken()`
```suggestion
		token, _, err := t.deleteToken(ctx, &secret, deleteValidation, options)
		if err != nil {
			return nil, apierrors.NewInternalError(fmt.Errorf("error deleting token %s: %w",
				secret.Name, err))
		}
```

Reply:
done, commit 955267bd9b9ad0e348e862eaded9e6b2f32a2b80